### DISCUSSION TITLE
Is there a "IsWasm" and "IsGtk" like property that can be used at compile time?

### DISCUSSION THREAD

**Discussion by igiona** [unoplatform/uno#12360] [Q&A]
In the default Directory.Build.props there is a set of IsX properties:

```
<IsAndroid>false</IsAndroid>
<IsIOS>false</IsIOS>
<IsMac>false</IsMac>
<IsMacCatalyst>false</IsMacCatalyst>
<IsWinAppSdk>false</IsWinAppSdk>
```

Is there a way to add support to IsWasm and IsGtk?
Can then these properties be used somehow as macro in the code, similarly to HAS_UNO ?

Many thanks in Advance

**Comment by jeromelaban** [comment]
You can use these compile time constants: https://platform.uno/docs/articles/platform-specific-csharp.html?q=partial#if-conditionals.

That being said, why do you need to make platform specific code at compile time? It's generally best to use runtime conditions.

**Reply by igiona** [reply]
Thanks for sharing.
I don't have a very specific reason why to choose compile vs runtime in this case.

What would be the recommended runtime check?
I thought of using `context.HostingEnvironment` but a part of the ApplicationName, I don't see a property/method that might help me here.
